### PR TITLE
Feature 168 report posts

### PR DESCRIPTION
**Pull request by John-Paul-Larkin**
# ✨ Codu Pull Request 💻

## Pull Request details:
This is the updated report comment feature . Should not require much review, as it basically all the same code form the previous PR. The only difference is the trpc router parts.

Need to add an ADMIN_EMAIL to the .env file.

### CODE REVIEW

**Comment by NiallJoeMaher** [comment]
I'll need to test this code a little since it'll result in emails to me 🙈 I'll try get to it later today! Thanks. 🦾

**Comment by John-Paul-Larkin** [comment]
No problem. There is a good few changes.

File: server/api/router/emailReport.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import toast from "react-hot-toast";
+import { SendEmailSchema } from "../../../schema/comment";
+import sendEmail from "@/utils/sendEmail";
+
+import { createTRPCRouter, protectedProcedure } from "../trpc";
+
+export const emailReportRouter = createTRPCRouter({
+  send: protectedProcedure
```
Comment (NiallJoeMaher) [inline server/api/router/emailReport.ts:8]: Do we tell a user to login if they want to report a post?

Comment (NiallJoeMaher) [reply server/api/router/emailReport.ts:8]: And if not maybe this will be an enhancement later. 🦾

Comment (John-Paul-Larkin) [reply server/api/router/emailReport.ts:8]: onClick={() => (session ? handleOpenModal() : signIn())}

I handled this on the button. If there is no session the auth signIn() funciton is called. 
Is that the right way to handle it?

File: components/Comments/ReportComments.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    };
+    document.addEventListener("mousedown", checkIfClickedOutside);
+    return () => {
+      document.removeEventListener("mousedown", checkIfClickedOutside);
+    };
+  }, [isModalOpen]);
+
+  return (
+    <>
+      <button
```
Comment (John-Paul-Larkin) [inline components/Comments/ReportComments.tsx:99]: Just figured out how to comment on a particular line :)

**Review by NiallJoeMaher — APPROVED** [review]
Nice one! I think I'll open up an issue to create one for blog posts too in case you want to take it 😉 Thanks a million! 🌮

**Comment by John-Paul-Larkin** [comment]
> Nice one! I think I'll open up an issue to create one for blog posts too in case you want to take it 😉 Thanks a million! 🌮

Blogs as in Articles? 
I had thought about it, but I wasnt sure. I could easily add the same flag button at the bottom of the article, just modify the props for the email. Or I could make it into an actual button, with label 'report blog'. Just to differentiate it.

BTW if you test the email and all goes well could you forward it to me, johnplarkin@gmail.com thanks

**Comment by NiallJoeMaher** [comment]
> > Nice one! I think I'll open up an issue to create one for blog posts too in case you want to take it 😉 Thanks a million! 🌮
> 
> 
> 
> Blogs as in Articles? 
> 
> I had thought about it, but I wasnt sure. I could easily add the same flag button at the bottom of the article, just modify the props for the email. Or I could make it into an actual button, with label 'report blog'. Just to differentiate it.
> 
> 
> 
> BTW if you test the email and all goes well could you forward it to me, johnplarkin@gmail.com thanks

I'll tag you in an issue later. I think we can use the 3 dots menu on the options bar and add "Report post" as one of the options from the menu. Simple and takes up no more space. In fact that might be nice pattern for anywhere we need it. 😅

**Pull request merged**